### PR TITLE
Updates to Symfony >= 2.5 components

### PR DESCRIPTION
**Pull request by lexxorlov**
I have done small fixes for new Symfony components with v. >= 2.5, 3. Fixed dialog -> question and command

### CODE REVIEW

**Comment by lexxorlov** [comment]
I did fixes in the github edit mode, thus so many commits :) But i have checked with Symfony 3.0 and i think it will work with Symfony >= 2.5. Because sf developers have changed this feature from 2.5. http://symfony.com/doc/current/components/console/helpers/dialoghelper.html

**Comment by garak** [comment]
You need to squash your commits. Follow the link provided in CONTRIBUTING.md file.
Your commits add some random spaces, and double pipes in composer.json.
Please fix this. Thank you.

**Comment by lexxorlov** [comment]
I have formatted composer.json. Fixed test command and helper.

**Comment by lexxorlov** [comment]
This is my tests log http://joxi.ru/DmBDkZ0uNgjX7A. But i have set github token.

File: src/Bowerphp/Command/InitCommand.php [hunk 1]
```diff
@@ -19,21 +19,22 @@
...
      */
     protected function configure()
     {
         $this
-            ->setName('init')
-            ->setDescription('Initializes a bower.json file')
-            ->setHelp(<<<EOT
+                ->setName('init')
+                ->setDescription('Initializes a bower.json file')
+                ->setHelp(<<<EOT
```
Comment (sensorario) [inline src/Bowerphp/Command/InitCommand.php]: I think your editor has 8 space formatting configuration.

**Comment by lexxorlov** [comment]
I use NetBeans. He should format correct.

**Comment by garak** [comment]
Still need to remove duplicate pipes in composer.json and squash.
Anyway, I'm afraid we cannot just drop support for Symfony 2.3, that is an LTS.
The correct way to support Symfony 3 is adding conditions, so we'll end up using both features (e.g. both Question and Dialog helper), using the correct one depending on the Symfony version.

**Comment by lexxorlov** [comment]
Can you explain what should i do with composer.json ? About Symfony version i understood. If i have some free time i will do it.

File: composer.json [hunk 2]
```diff
@@ -26,22 +26,24 @@
         "php": ">=5.3.6",
         "ext-zip": "*",
         "ext-fileinfo": "*",
-        "symfony/console": "~2.3",
-        "symfony/finder": "~2.2",
-        "symfony/filesystem": "~2.1",
-        "symfony/process": "~2.1",
+        "symfony/console": ">=2.5||~3.0",
```
Comment (garak) [inline composer.json]: ">=2.5||~3.0" should be ">=2.5|~3.0"

Comment (clue) [inline composer.json]: The `>=2.5` part would mean it's also compatible with 3.0 and a future 4.0.

It we keep compatibility with previous releases, this should probably be something like `~2.3 || ~3.0` or `^2.3 || ^3.0`, see e.g. composer itself: https://github.com/composer/composer/blob/876f14341814d427e87dd358ed91ccc855678a5c/composer.json#L32

Comment (garak) [inline composer.json]: Agreed. Should be "~2.3||~3.0" (or "~2.3|~3.0", I still need to understand the question about single or double pipe)

**Comment by lexxorlov** [comment]
Added support  for Symfony 2.3 and Symfony 3

File: src/Bowerphp/Command/Helper/DialogHelper.php [hunk 3]
```diff
@@ -24,8 +24,8 @@ class DialogHelper extends BaseDialogHelper
      *  "Do you want to continue [yes]:"
      *
      * @param string $question The question you want to ask
-     * @param mixed  $default  Default value to add to message, if false no default will be shown
-     * @param string $sep      Separation char for between message and user input
+     * @param mixed $default Default value to add to message, if false no default will be shown
```
Comment (garak) [inline src/Bowerphp/Command/Helper/DialogHelper.php]: What is this? Did you run php-cs-fixer?

File: src/Bowerphp/Command/InitCommand.php [hunk 4]
```diff
@@ -13,12 +13,14 @@
 
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\HttpKernel\Kernel;
```
Comment (garak) [inline src/Bowerphp/Command/InitCommand.php]: Too bad HttpKernel is **not** in our dependencies.
You should rely on feature detect, not on declared versions.

Comment (lexxorlov) [inline src/Bowerphp/Command/InitCommand.php]: Sorry, it is my error. I will fix it.

File: src/Bowerphp/Command/Helper/DialogHelper.php (hunk 3 above)
Comment (lexxorlov) [inline src/Bowerphp/Command/Helper/DialogHelper.php]: I think this is my IDE code formatter

**Comment by lexxorlov** [comment]
Was removed "kernel" dependency. Now it is simple class checker.

**Comment by garak** [comment]
@lexxorlov you need to clone this repository on your machine and work from that.
Otherwise, you cannot run tests nor execute php-cs-fixer on the code, nor squash your commits.
All these are required, otherwise I'll forced to reject your PR. Thank you.

**Comment by lexxorlov** [comment]
I have done this. I have fixed all your remarks.

**Comment by lexxorlov** [comment]
Will you merge PR?

**Comment by garak** [comment]
I still see 24 commits in this PR, so it's clear that you didn't squash.
Also, I see blank lines, uncorrect spaces, so it's clear that you didn't run php-cs-fixer.

**Comment by lexxorlov** [comment]
Commits were merged. 
php-cs-fixer was done.

**Comment by garak** [comment]
Thanks!

**Comment by garak** [comment]
Just to know... are your tests passing? Mine are not :disappointed:

**Comment by garak** [comment]
```
PHP Fatal error:  Class 'Bowerphp\Command\Helper\QuestionHelper' not found in /home/garak/prj/bowerphp/src/Bowerphp/Console/Application.php on line 130
Fatal error: Class 'Bowerphp\Command\Helper\QuestionHelper' not found in /home/garak/prj/bowerphp/src/Bowerphp/Console/Application.php on line 130
``

**Comment by lexxorlov** [comment]
Yes, i have checked. But i forgot add to git one file :( sorry for this. If you want i can provide screen-shots with tests :)

**Pull request merged**